code/modules/reagents/reagent_containers/hypospray.dm

```diff
@@ -138,10 +138,28 @@
 
 /obj/item/reagent_containers/hypospray/medipen/examine()
 	. = ..()
-	if(reagents?.reagent_list.len)
-		. += span_notice("It is currently loaded.")
+	if(length(reagents?.reagent_list))
+		var/reagent_text = ""
+
+		if(length(reagents.reagent_list) == 1)
+			var/datum/reagent/first = reagents.reagent_list[1]
+			reagent_text = first.name
+
+		else
+			var/len = length(reagents.reagent_list)
+			var/list/reagent_names_list = list()
+			for(var/i in 1 to (len - 2))
+				var/datum/reagent/reagent_object = reagents.reagent_list[i]
+				reagent_names_list += "[reagent_object.name], "
+			var/datum/reagent/last_reagent = reagents.reagent_list[len]
+			var/datum/reagent/second_to_last_reagent = reagents.reagent_list[len - 1]
+			reagent_names_list += "[second_to_last_reagent.name], and [last_reagent.name]"
+			reagent_text = reagent_names_list.Join("")
+
+		. += span_notice("There\'s a small LCD screen on the side of the medipen which reads, 'WARNING: This medipen contains [reagent_text]. Do not use if allergic to any listed chemicals.' in small text.") //cheapskates at nanotrasen couldnt ball out for LED
```

Comment:
```suggestion
	if(length(reagents?.reagent_list))
		. += span_notice("There\'s a small LCD screen on the side of the medipen which reads, 'WARNING: This medipen contains [pretty_string_from_reagent_list(reagents.reagent_list)]. Do not use if allergic to any listed chemicals.' in small text.") //cheapskates at nanotrasen couldnt ball out for LED
```

Reply:
no fucking way thats all I had to do

Reply:
im testing the code right now I swear to god

Reply:
While I am infinitely angered at the fact that this worked... I just want a list of the names of the chemicals... You do not understand my emotion right now it's just pure anger

Reply:
```
	if(length(reagents?.reagent_list))
		. += span_notice("There\'s a small LCD screen on the side of the medipen which reads, 'WARNING: This medipen contains [pretty_string_from_reagent_list(reagents.reagent_list, TRUE, ", ", TRUE)]. Do not use if allergic to any listed chemicals.' in small text.") //cheapskates at nanotrasen couldnt ball out for LED

/proc/pretty_string_from_reagent_list(list/reagent_list, names_only, join_text = " | ", final_and)
	//Convert reagent list to a printable string for logging etc
	var/list/rs = list()
	var/list_length = reagent_list.len
	for (var/datum/reagent/R in reagent_list)
		list_length--
		if(final_end && list_length == 1)
			rs += "and [R.name][names_only ? null : ", [R.volume]"]"
		else
			rs += "[R.name][names_only ? null : ", [R.volume]"]"


	return rs.Join(join_text)
```

lets adapt the proc itself to be a bit more robust in that case.

Reply:
But why? It accomplishes the same thing as the current code, albeit a microscopic optimization, I just don't see a reason to